### PR TITLE
doc/rados/configuration/msgr2: some documentation about msgr2

### PR DESCRIPTION
**Pull request by liewegas**
This doesn't integrate very well into network-config.rst, mostly because
that document is horribly out of data and I don't know where to start.
:(

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: doc/rados/configuration/msgr2.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+Prior to nautilus, all network addresses were rendered like
+``1.2.3.4:567/89012`` where there was an IP address, a port, and a
+nonce to uniquely identify a client or daemon on the network.
+Starting with nautilus, we now have three different address types:
+
+* **v2**: ``v2:1.2.3.4:578/89012`` identifies a daemon binding to a
+  port speaking the new v2 protocol
+* **v1**: ``v1:1.2.3.4:578/89012`` identifies a daemon binding to a
+  port speaking the legacy v2 protocol.  Any address that was
```
Comment (xiexingguo) [inline doc/rados/configuration/msgr2.rst]: s/v2/v1/

File: doc/rados/configuration/msgr2.rst [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  previously shown with any prefix is now shows as a ``v1:`` address.
+* **TYPE_ANY** addresses identify a client that can speak either
+  version of the protocol.  Prior to nautilus, clients would appear as
+  ``1.2.3.4:0/123456``, where the port of 0 indicates they are clients
+  and do not accept incoming connections.  Starting with Nautlius,
+  these clients are now internally represented by a **TYPE_ANY**
+  address, and still shown with no prefix, because they can may
```
Comment (xiexingguo) [inline doc/rados/configuration/msgr2.rst]: extra `may`

File: doc/rados/configuration/msgr2.rst [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  cluster.
+
+There are a parallel set of options that apply specifically to
+monitors, allowing administrators to set different (usually more
+secure) requirements on communication with the monitors.
+
+* ``ms_mon_cluster_mode`` is the connection mode (or permitted modes)
+  to use between monitors.
+* ``ms_mon_service_mode`` is a list of permitted modes for clients or
+  other Ceph daemons to use when connection to monitors.
```
Comment (xiexingguo) [inline doc/rados/configuration/msgr2.rst]: s/connection/connecting/

**Comment by xiexingguo** [comment]
This looks great and I wish every newly added feature could have a such exhaustive documentation.

File: doc/rados/configuration/msgr2.rst [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+* **v2**: ``v2:1.2.3.4:578/89012`` identifies a daemon binding to a
+  port speaking the new v2 protocol
+* **v1**: ``v1:1.2.3.4:578/89012`` identifies a daemon binding to a
+  port speaking the legacy v2 protocol.  Any address that was
+  previously shown with any prefix is now shows as a ``v1:`` address.
+* **TYPE_ANY** addresses identify a client that can speak either
+  version of the protocol.  Prior to nautilus, clients would appear as
+  ``1.2.3.4:0/123456``, where the port of 0 indicates they are clients
+  and do not accept incoming connections.  Starting with Nautlius,
```
Comment (neha-ojha) [inline doc/rados/configuration/msgr2.rst]: s/Nautlius/Nautilus/

File: doc/rados/configuration/msgr2.rst [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+What is it
+----------
+
+The messenger v2 protocol, or msgr2, is the second major revision on
+Ceph's on-wire protocol.  It brings with it several key features:
+
+* A *secure* mode that encrypts all data passing over the network
+* Improved encapslation of authentication payloads, enabling future
+  integration of new authentication modes like Kerberos
+* Improved earlier feature advertisement and negotiatiation, enabling
```
Comment (neha-ojha) [inline doc/rados/configuration/msgr2.rst]: s/negotiatiation/negotiation/

File: doc/rados/configuration/msgr2.rst [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+============
+
+What is it
+----------
+
+The messenger v2 protocol, or msgr2, is the second major revision on
+Ceph's on-wire protocol.  It brings with it several key features:
+
+* A *secure* mode that encrypts all data passing over the network
+* Improved encapslation of authentication payloads, enabling future
```
Comment (neha-ojha) [inline doc/rados/configuration/msgr2.rst]: s/encapslation/encapsulation/

File: doc/rados/configuration/msgr2.rst [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+Prior to nautilus, all network addresses were rendered like
+``1.2.3.4:567/89012`` where there was an IP address, a port, and a
+nonce to uniquely identify a client or daemon on the network.
+Starting with nautilus, we now have three different address types:
+
+* **v2**: ``v2:1.2.3.4:578/89012`` identifies a daemon binding to a
+  port speaking the new v2 protocol
+* **v1**: ``v1:1.2.3.4:578/89012`` identifies a daemon binding to a
+  port speaking the legacy v2 protocol.  Any address that was
+  previously shown with any prefix is now shows as a ``v1:`` address.
```
Comment (neha-ojha) [inline doc/rados/configuration/msgr2.rst]: is now s/shows/shown/

**Comment by neha-ojha** [comment]
jenkins render docs

**Comment by neha-ojha** [comment]
http://docs.ceph.com/ceph-prs/26867/rados/configuration/msgr2/

**Pull request merged**